### PR TITLE
tests: sensor SBS EMUL_CONFIG to testcase.yaml

### PR DESCRIPTION
**Pull request by aaronemassey**
NOTE: This currently fails because CONFIG_EMUL isn't being applied for some reason. I've found this to work with other tests and I'm not sure where to go next on debugging this so I'm pushing this WIP PR for folks to examine, namely @galak .

There exists several <board>.conf files in the boards directory of the sbs_gauge tests. All these confs do is enable CONFIG_EMUL. We can remove these board confs in favor of a single extra_configs entry in the sensor sbs gauge testcase.yaml file.

Remove the sensor test_bs_gauge test's board conf files in favor of enabling CONFIG_EMUL in the testcase.yaml file.

### CODE REVIEW

**Comment by aaronemassey** [comment]
https://github.com/zephyrproject-rtos/zephyr/actions/runs/4834505321/jobs/8615815150?pr=57396#step:13:642 is indicative of what's happening locally as well.

**Comment by galak** [comment]
> https://github.com/zephyrproject-rtos/zephyr/actions/runs/4834505321/jobs/8615815150?pr=57396#step:13:642 is indicative of what's happening locally as well.

seems like something isn't getting set in the .config.  Maybe compare before and after this change to see what the .config looks like.

**Comment by aaronemassey** [comment]
@galak 

Figured out the problem. It was actually really obvious when I saw that the failing emulated builds were happening because we were doing emulated builds non-emulated compatible. I just had missed it entirely.

Digging further into it. I realized `CONFIG_EMUL` is entirely redundant and should just be used as a helper so I added https://github.com/zephyrproject-rtos/zephyr/pull/57568, but that's a separate concern.

File: tests/drivers/sensor/sbs_gauge/testcase.yaml [hunk 1]
```diff
@@ -3,7 +3,8 @@ tests:
   drivers.sensors.sbs_gauge:
     build_only: true
     tags: drivers sensors
-    filter: dt_compat_enabled("sbs,sbs-gauge")
+    filter:
+      dt_compat_enabled("sbs,sbs-gauge") and not dt_compat_enabled("zephyr,i2c-emul-controller")
```
Comment (aaronemassey) [inline tests/drivers/sensor/sbs_gauge/testcase.yaml:7]: Sorry for the rebase push. This was the only actual change in this PR.

**Comment by aaronemassey** [comment]
Waiting for https://github.com/zephyrproject-rtos/zephyr/pull/57568 to merge first.

**Review by asemjonovs — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**